fix: fix contract mismatches in ESG BFF

nsprenkle opened the pull request:
## Description

- Change `submissionUuid` to `submissionUUID` in `initialize` call to match contract
- Move `rubricConfig` inside of `oraMetadata` to match contract
- Refactor how we get rubric info, pull from ORA block directly instead of using rubric reuse API
- Many, many test updates

FYI @openedx/masters-devs-gta 

## Testing instructions

Do it work tho?

nsprenkle committed:
fix: update capitalizations in submissionUUID

nsprenkle committed:
fix: move rubricConfig inside of oraMetadata

mattcarter reviewed (approved):
lgtm

jansenk commented on lms/djangoapps/ora_staff_grader/serializers.py:
```diff
@@ -144,14 +144,18 @@ class InitializeSerializer(serializers.Serializer):
-    rubricConfig = RubricConfigSerializer()
+
+    def to_representation(self, instance):
+        """ Move rubric config inside of ORA metadata """
```
I'm very hesitant to manually touch to_representation, and there is a way to make this serializer-y, but it's kinda ugly.

if you make 145 `oraMetadata = OpenResponseMetadataSerializer(source='*')` the it will pass "this" object to that serializer. 

Then we can just add a field to that serializer 
  `rubricConfig = RubricConfigSerializer()`
  but then would have to modify the existing field lookups to go one additional level down (`name = serializers.CharField(source='display_name')` becomes     `name = serializers.CharField(source='oraMetadata.display_name')` for example)

nsprenkle replied:
I think the "right" way to do this is to transform the rubric info right from the ORA to remove having to call the rubric API (but we were leveraging the fact that the rubric API already creates the shape and (mostly) the names we wanted).

Any other way is kind of gross because this info is nested differently than we want, your method is just slightly less gross.

jansenk replied:
We could move the rubric to the context and then have a MethodField in orametadata that just pulls that out of context?

jansenk replied:
I don't know I think it makes most sense to have it separate and that this should be a frontend change but whatever

nsprenkle committed:
refactor: get rubricConfig from ORA directly

Instead of getting rubric data from an XBlock handler, pull directly
from the source ORA block. This duplicates some data from ORA rubric
reuse API but removes an API call and just makes more sense with how our
naming paths are diverging between the implementations

nsprenkle committed:
fix: add default rubric criterion feedback setting

nsprenkle committed:
test: update view tests to remove rubric API call

nsprenkle committed:
test: update tests and test data

nsprenkle committed:
chore: remind me again why we enforce formatting?
